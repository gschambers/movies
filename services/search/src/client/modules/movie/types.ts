export interface Movie {
    id: number
    title: string;
    imageUrl: string | null;
    rating: number;
}
