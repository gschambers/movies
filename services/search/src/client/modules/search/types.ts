export interface SearchResult<T> {
    results: T[];
}
