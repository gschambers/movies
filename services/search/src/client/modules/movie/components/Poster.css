.Poster {
    position: relative;
}

.Poster--small {
    width: 200px;
    height: 300px;
}

.Poster--medium {
    width: 300px;
    height: 450px;
}

.Poster--large {
    width: 450px;
    height: 675px;
}

    .Poster-backdrop {
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        background-size: cover;
        border-radius: 5px;
        box-shadow: 0 16px 32px hsl(220, 21%, 8%);
    }

    .Poster-strapline {
        display: flex;
        position: absolute;
        left: 5px;
        right: 5px;
        color: white;
    }

    .Poster--small .Poster-strapline {
        bottom: -25px;
        font-size: 13px;
    }

    .Poster--medium .Poster-strapline {
        bottom: -30px;
        font-size: 16px;
    }

    .Poster--large .Poster-strapline {
        bottom: -35px;
        font-size: 20px;
    }

        .Poster-rating {
            margin-left: 5px;
            font-weight: 300;
            color: hsl(220, 21%, 75%);
        }
