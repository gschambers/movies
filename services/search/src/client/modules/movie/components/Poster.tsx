import * as React from "react";
import { Movie } from "../types";

import "./Poster.css";

interface Props {
    movie: Movie;
    showStrapline?: boolean;
    size?: "small" | "medium" | "large";
}

export const Poster: React.SFC<Props> = ({ movie, showStrapline = true, size = "medium" }) => (
    <div className={`Poster Poster--${size}`}>
        <div className="Poster-backdrop" style={{ backgroundImage: `url(/images${movie.imageUrl})` }} />
        {showStrapline && (
            <div className="Poster-strapline">
                <div className="Poster-title">{movie.title}</div>
                <div className="Poster-rating">({movie.rating})</div>
            </div>
        )}
    </div>
);
