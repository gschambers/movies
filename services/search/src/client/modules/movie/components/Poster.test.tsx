import * as React from "react";
import * as renderer from "react-test-renderer";

import { Poster } from "./Poster";
import { Movie } from "../types";

describe("Poster", () => {
    describe("given a movie as input", () => {
        it("should render backdrop and title correctly", () => {
            const movie: Movie = {
                id: 153,
                title: "Lost in Translation",
                imageUrl: "/5T8VvuFTdaawKLJk34i69Utaw7o.jpg",
                rating: 7.4,
            };

            const poster = renderer
                .create(<Poster movie={movie} />)
                .toJSON();

            expect(poster).toMatchSnapshot();
        });
    });
});
