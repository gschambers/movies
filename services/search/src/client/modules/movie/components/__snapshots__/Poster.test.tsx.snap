// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster given a movie as input should not render title and rating, if specified 1`] = `
<div
  className="Poster Poster--medium"
>
  <div
    className="Poster-backdrop"
    style={
      Object {
        "backgroundImage": "url(/images/5T8VvuFTdaawKLJk34i69Utaw7o.jpg)",
      }
    }
  />
</div>
`;

exports[`Poster given a movie as input should render backdrop and title correctly 1`] = `
<div
  className="Poster Poster--medium"
>
  <div
    className="Poster-backdrop"
    style={
      Object {
        "backgroundImage": "url(/images/5T8VvuFTdaawKLJk34i69Utaw7o.jpg)",
      }
    }
  />
  <div
    className="Poster-strapline"
  >
    <div
      className="Poster-title"
    >
      Lost in Translation
    </div>
    <div
      className="Poster-rating"
    >
      (
      7.4
      )
    </div>
  </div>
</div>
`;
